corehq/apps/couch_sql_migration/management/commands/couch_sql_diff.py

```diff
@@ -277,9 +278,13 @@ def diff(couch_json, sql_json):
             couch_case["domain"], couch_case['_id'], None, save=False, lock=False
         ).to_json()
         diffs = diff(couch_case, sql_json)
-        if diffs and should_sort_sql_transactions(sql_case, couch_case):
-            sql_case = rebuild_case_with_couch_action_order(sql_case)
-            diffs = diff(couch_case, sql_case.to_json())
+        if diffs:
+            if should_sort_sql_transactions(sql_case, couch_case):
+                sql_case = rebuild_case_with_couch_action_order(sql_case)
+                diffs = diff(couch_case, sql_case.to_json())
+            elif not was_rebuilt(sql_case):
+                sql_case = rebuild_case(sql_case)
```

Comment:
Just curious, in this case what good would rebuilding do? Is it just like a "maybe this will help in 5% of cases" kind of thing, or is it targeting a particular known issue?

Reply:
It handles a real scenario that happens with some frequency, although I haven't measured exactly how often. Something with the way cases are built by the migration isn't always perfect. Maybe I should research that more to figure out the precise reason since all I have right now is a hand-wavy explanation.